presto-docs/src/main/sphinx/security/internal-communication.rst

```diff
@@ -112,3 +112,38 @@ To enable SSL/TLS for Presto internal communication, do the following:
 
         internal-communication.https.keystore.path=<keystore path>
         internal-communication.https.keystore.key=<keystore password>
+
+
+Enabling encryption introduces performance impact. Performance degradation can
+vary based on the environment, queries, concurrency.
+
+For the queries, that are not require to much data to be transferred between the Presto
+nodes (such as simple `SELECT count(*) FROM table`), performance
+impact is unnoticeable.
+
+However for the queries that are CPU intensive, and require considerable amount of data
+to be transferred between the nodes (for example when repartitioning is needed:
+JOIN query with distributed join enabled, or queries with GROUP BY), performance impact
```

Comment:
"JOIN query with distributed join enabled, or queries with GROUP BY" -> distributed joins, aggregations, and windowing functions